lib/Doctrine/ORM/Mapping/DefaultTypedFieldMapper.php

```diff
@@ -36,12 +57,47 @@ final class DefaultTypedFieldMapper implements TypedFieldMapper
         'string' => Types::STRING,
     ];
 
+    /** It maps built-in Doctrine types to PHP types */
+    private const BUILTIN_TYPES_MAP = [
+        AsciiStringType::class => 'string',
+        BigIntType::class => 'string',
+        BooleanType::class => 'bool',
+        DateType::class => DateTime::class,
+        DateImmutableType::class => DateTimeImmutable::class,
+        DateIntervalType::class => DateInterval::class,
+        DateTimeType::class => DateTime::class,
+        DateTimeImmutableType::class => DateTimeImmutable::class,
+        DateTimeTzType::class => DateTime::class,
+        DateTimeTzImmutableType::class => DateTimeImmutable::class,
+        DecimalType::class => 'string',
+        FloatType::class => 'float',
+        GuidType::class => 'string',
+        IntegerType::class => 'int',
+        JsonType::class => 'array',
+        SimpleArrayType::class => 'array',
+        SmallIntType::class => 'int',
+        StringType::class => 'string',
+        TextType::class => 'string',
+        TimeType::class => DateTime::class,
+        TimeImmutableType::class => DateTimeImmutable::class,
+    ];
+
     /** @param array<class-string|ScalarName, class-string<Type>|string> $typedFieldMappings */
     public function __construct(array $typedFieldMappings = [])
     {
         $this->typedFieldMappings = array_merge(self::DEFAULT_TYPED_FIELD_MAPPINGS, $typedFieldMappings);
     }
 
+    /**
+     * @internal
+     *
+     * @param class-string<Type> $typeName
+     */
+    public function getBuiltInType(string $typeName): string
+    {
+        return self::BUILTIN_TYPES_MAP[$typeName] ?? $typeName;
+    }
```

Comment:
Perhaps, instead of exposing the type, we could ask a boolean question `typeConformsType()` or such?

Hard to say in this scope, so the naming is a bit wonky, but just an idea.

Reply:
Perhaps the best approach remains to keep this `private` in the validator itself, for now, rather than having it in `DefaultTypedFieldMapper`